Add github discussions link

nzw0301 opened the pull request:
<!-- Thank you for creating a pull request! In general, we merge your pull request after it gets two or more approvals. To proceed to the review process by the maintainers, please make sure that the PR meets the following conditions: (1) it passes all CI checks, and (2) it is neither in the draft nor WIP state. If you wish to discuss the PR in the draft state or need any other help, please mention the Optuna development team in the PR. -->

## Motivation
<!-- Describe your motivation why you will submit this PR. This is useful for reviewers to understand the context of PR. -->

I'm not sure when Optuna starts to use [github discussions](https://github.com/optuna/optuna/discussions), but already there is a discussion on it. So it might be useful to mention it from `README.md`.

## Description of the changes
<!-- Describe the changes in this PR. -->

By following other communication channels, this PR adds a link to optuna/optuna's github discussions page.

toshihikoyanase commented on README.md:
```diff
@@ -150,10 +150,12 @@ Also, we also provide Optuna docker images on [DockerHub](https://hub.docker.com
 
 ## Communication
 
+- [GitHub Discussions] for feature requests and questions.
```
I didn't know that we opened the GitHub discussions, but users have already communicates each other on it 👍 . BTW, Both `GitHub Discussions` and `GItHub Issues` have `feature requests` and `questions`, and users might get confused. So, we may need to discuss the policy of these tools before adding these sentences.

c-bata replied:
We discussed how we should use GitHub Discussions at the Optuna core development team MTG. And then we decided to use Github Discussions as a place where users can casually post questions, and GitHub issues will continue to be used for bug reports and feature requests.

Do you think it makes sense?

c-bata replied:
So we might have to update Optuna's issue template as well.
https://github.com/optuna/optuna/blob/master/.github/ISSUE_TEMPLATE/questions-help-support.yml

FYI: In Optuna Dashboard, I prepared an issue template to prevent users to ask support questions.
https://github.com/optuna/optuna-dashboard/blob/main/.github/ISSUE_TEMPLATE/question.md

nzw0301 replied:
@c-bata thank you for sharing the discussion. Yes, I think it makes sense.

nzw0301 commented:
I've updated `README.md`'s description and replaced the Q&A issue template with redirection to discussions by following Optuna-dashboard's issue template.

c-bata reviewed (approved):
Thanks for the update. LGTM!

toshihikoyanase commented on .github/ISSUE_TEMPLATE/questions-help-support.yml:
GitHub shows some problems with this template. Could you check the following link?
https://github.com/optuna/optuna/blob/65bbc11e1677f70f41f0c33bce7cbc3f3e8577e9/.github/ISSUE_TEMPLATE/questions-help-support.yml

![image](https://user-images.githubusercontent.com/3255979/174592287-6cdec5ee-b515-4bd2-9aa4-e89f55714420.png)

nzw0301 replied:
Oh, i didn't notice them! Thanks! I've removed all warnings!

toshihikoyanase replied:
Thank you for your update. I confirmed that they were resolved.

nzw0301 commented on .github/ISSUE_TEMPLATE/questions-help-support.yml:
This field is still necessary to avoid [this]( https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#body-must-contain-at-least-one-non-markdown-field).

toshihikoyanase reviewed (approved):
Thank you for your update. LGTM!
